ts/src/p2b.ts

```diff
@@ -0,0 +1,1263 @@
+        return this.parseTrades (result, market, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        //
+        // fetchTrades
+        //
+        //    {
+        //        id: '7495738622',
+        //        type: 'sell',
+        //        time: '1699255565.445418',
+        //        amount: '252.6',
+        //        price: '0.3422'
+        //    }
+        //
+        // fetchMyTrades
+        //
+        //    {
+        //        "deal_id": 7450617292,              // Deal id
+        //        "deal_time": 1698506956.66224,      // Deal execution time
+        //        "deal_order_id": 171955225751,      // Deal order id
+        //        "opposite_order_id": 171955110512,  // Opposite order id
+        //        "side": "sell",                     // Deal side
+        //        "price": "0.05231",                 // Deal price
+        //        "amount": "0.002",                  // Deal amount
+        //        "deal": "0.00010462",               // Total (price * amount)
+        //        "deal_fee": "0.000000188316",       // Deal fee
+        //        "role": "taker",                    // Role. Taker or maker
+        //        "isSelfTrade": false                // is self trade
+        //    }
+        //
+        // fetchOrderTrades
+        //
+        //    {
+        //        "id": 7429883128,             // Deal id
+        //        "time": 1698237535.41196,     // Deal execution time
+        //        "fee": "0.01755848704",       // Deal fee
+        //        "price": "34293.92",          // Deal price
+        //        "amount": "0.00032",          // Deal amount
+        //        "dealOrderId": 171366551416,  // Deal order id
+        //        "role": 1,                    // Deal role (1 - maker, 2 - taker)
+        //        "deal": "10.9740544"          // Total (price * amount)
+        //    }
+        //
+        const timestamp = this.safeIntegerProduct2 (trade, 'time', 'deal_time', 1000);
+        let takerOrMaker = this.safeString (trade, 'role');
+        if (takerOrMaker === '1') {
+            takerOrMaker = 'maker';
+        } else if (takerOrMaker === '2') {
+            takerOrMaker = 'taker';
+        }
+        return this.safeTrade ({
+            'info': trade,
+            'id': this.safeString2 (trade, 'id', 'deal_id'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': this.safeString (market, 'symbol'),
+            'order': this.safeString2 (trade, 'dealOrderId', 'deal_order_id'),
+            'type': undefined,
+            'side': this.safeString (trade, 'type'),
+            'takerOrMaker': takerOrMaker,
+            'price': this.safeString (trade, 'price'),
+            'amount': this.safeString (trade, 'amount'),
+            'cost': this.safeString (trade, 'deal'),
+            'fee': {
+                'currency': market['base'],
```

Comment:
you sure this is the case for both sides? (sell and buy)

Reply:
It's always the quote currency